### ISSUE TITLE
Add deletion lock on long running test resources

### ISSUE THREAD

**Issue by kachawla** [radius-project/radius#7719]
Long running test cluster was deleted by mistake recently. We should leverage [resource locks](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/lock-resources?tabs=json) to prevent accidental deletion.

[AB#12632](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/12632)

**Comment by sylvainsf** [comment]
Potentially stacks could be a better idea, we should explore.

**Comment by nicolejms** [comment]
We should add something to the deployment to lock it

**Comment by kachawla** [comment]
> We should add something to the deployment to lock it

Yeah the goal here is to set the lock in the bicep template that we use to deploy LRT infra https://github.com/radius-project/radius/blob/main/test/infra/azure/main.bicep

**Comment by kachawla** [comment]
> Potentially stacks could be a better idea, we should explore.

Interesting, could you say more on how stacks would help here? This motivation behind this was accidental deletion of LRT resources.
